### PR TITLE
Added Ellipsoid and Capsule shapes to TPE

### PR DESCRIPTION
**Pull request by ahcorde**
Added ellipsoid and capsule to TPE. This PR builds on top of this other PR https://github.com/ignitionrobotics/ign-physics/pull/202

Related with https://github.com/ignitionrobotics/ign-physics/issues/196 and https://github.com/ignitionrobotics/ign-physics/issues/195

It's a draft for now, I need to include some more tests.

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by ahcorde** [comment]
I opened this PR for review, I can add more tests if required

File: tpe/lib/src/Shape.hh [hunk 1]
```diff
@@ -163,6 +212,38 @@ class IGNITION_PHYSICS_TPELIB_VISIBLE CylinderShape : public Shape
...
+  /// \param[in] _other shape to copy from
+  public: Shape &operator=(const Shape &_other);
+
+  /// \brief Get ellipsoid radius
+  /// \return ellipsoid radius
+  public: math::Vector3d GetRadii() const;
+
+  /// \brief Set ellipsoid radius
+  /// \param[in] _radius ellipsoid radius
+  public: void SetRadii(math::Vector3d _radii);
```
Comment (chapulina) [inline tpe/lib/src/Shape.hh:238]:
```suggestion
  public: void SetRadii(const math::Vector3d &_radii);
```

File: tpe/plugin/worlds/shapes.world [hunk 2]
```diff
@@ -96,5 +96,36 @@
         </visual>
       </link>
     </model>
+
+    <model name="ellipsoid">
```
Comment (chapulina) [inline tpe/plugin/worlds/shapes.world:100]: Should we add a capsule too?

**Review by chapulina — COMMENTED** [review]
> It's a draft for now, I need to include some more tests.

I wasn't sure about the status of this PR, so I went ahead and gave it a quick review. It's looking good to me, I think it would be nice to add tests to `CollisionDetector_TEST`.

---

For future reference, I think it's also worth mentioning that TPE pretty much treats all shapes as boxes for the purpose of collision checking, because it only checks their AABBs. But Having these shapes supported means that worlds using complex shapes with other engines will work reasonably with TPE too.

**Comment by chapulina** [comment]
The Homebrew and Ubuntu warnings are not introduced by this PR, but the Windows warnings are

**Comment by ahcorde** [comment]
@chapulina fixed windows warnings

File: tpe/lib/src/CollisionDetector_TEST.cc [hunk 3]
```diff
@@ -240,6 +297,39 @@ TEST(CollisionDetector, CheckCollisions)
   EXPECT_EQ(8u, contacts.size());
   contacts = cd.CheckCollisions(entities, true);
   EXPECT_EQ(1u, contacts.size());
+
+  entities[modelC->GetId()] = modelC;
+  // collision between model E and B but not model A, C, D
```
Comment (chapulina) [inline tpe/lib/src/CollisionDetector_TEST.cc:302]: Can you add a test case where `D`, the capsule, collides with something?

File: tpe/plugin/src/ShapeFeatures.cc [hunk 4]
```diff
@@ -95,6 +95,83 @@ Identity ShapeFeatures::CastToCylinderShape(const Identity &_shapeID) const
...
+  // return invalid height if no collision found
+  return -1.0;
+}
+
+/////////////////////////////////////////////////
+Identity ShapeFeatures::AttachCapsuleShape(
+  const Identity &_linkID,
+  const std::string &_name,
+  const double _radius,
+  const double _height,
```
Comment (chapulina) [inline tpe/plugin/src/ShapeFeatures.cc:154]:
```suggestion
  const double _length,
```

**Review by chapulina — APPROVED** [review]
LGTM, I just have some minor final comments.

**Pull request merged**